WordPress/Sniffs/WP/I18nSniff.php

```diff
@@ -80,6 +80,8 @@ class I18nSniff extends AbstractFunctionRestrictionsSniff {
 		[bcdeEfFgGosuxX]           # Type specifier.
 	)/x';
 
+	const WRAPPED_IN_HTML_REGEX = '#^(<[^>]+>).*(<[^>]+>)$#';
```

Comment:
I agree with @chriskmnds that this regex needs improvement.

Some suggestions:
* Don't remember what doesn't need to be remembered.
* Match the (first part of the) content matched in the first tag with the content in the last tag.
* Keep in mind that HTML tags can have attributes. The regex in combination with the code below as is, would break straight away on this.
    ```php
    __( '<div class="my-class">Translatable content</div>', 'domain' );
    ```
* Keep in mind that there may be placeholders in the HTML attributes. If so, the HTML should not be removed and the warning not thrown.
    Without making the regex overly complicated, you could just check the full tag results of the tag at the start of the string with the same after running it through the `SPRINTF_PLACEHOLDER_REGEX`. If these are not the same, again, don't throw the message.
    ```php
    __( '<option id="%s" value="%s">Translatable option name</option>', 'domain' );
    ```

Let me know if you need help with this. I'm happy to provide you with a regex.

Reply:
Thanks, those are great points. I'll try tweaking the regex a bit.

Reply:
43c80e3